hikari/api/cache.py

```diff
@@ -54,8 +54,18 @@ class CacheView(typing.Mapping[_KeyT, _ValueT], abc.ABC):
     __slots__: typing.Sequence[str] = ()
 
     @abc.abstractmethod
-    def get_item_at(self, index: int) -> _ValueT:
-        """Get an entry in the view at position `index`."""
+    @typing.overload
+    def get_item_at(self, index: int, /) -> _ValueT:
+        ...
+
+    @abc.abstractmethod
```

Comment:
Does anyone happen to know whether this is valid? I mean, it passes type checking and the tests don't error but... it just kind of feels wrong.

Reply:
We have been doing it already. I don't know what the correct way of doing it is, but we have using this way already. https://github.com/hikari-py/hikari/blob/master/hikari/api/rest.py#L496-L587